events/events.go

```diff
@@ -343,7 +343,26 @@ func (b *BaseBody) SetIPAddress(ip string) {
 // OperationProvisioned represents a provision operation event.
 type OperationProvisioned struct {
 	BaseBody
-	Data OperationProvisionedData `json:"data"`
+	Data *OperationProvisionedData `json:"data"`
+}
+
+// OwnerData represents the owner of the event reference.
+type OwnerData struct {
+	UserID *manifold.ID `json:"user_id,omitempty"`
+	User   *User        `json:"user,omitempty"`
+
+	TeamID *manifold.ID `json:"team_id,omitempty"`
+	Team   *Team        `json:"team,omitempty"`
+}
+
+// SetOwner sets the owner depending if it is a user or a team.
+func (o *OwnerData) SetOwner(id manifold.ID) {
+	switch id.Type() {
+	case idtype.User:
+		o.UserID = &id
+	case idtype.Team:
+		o.TeamID = &id
+	}
```

Comment:
Just wondering if we should panic here if it's neither of these options? I'd rather have it panic when we give it an ID that isn't supposed to be passed through than having data missing.

Reply:
(If you do, add a comment about that in the comment ;-) )

Reply:
sure `panic("danger will robinson")`